components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/filter/CaptchaFilter.java

```diff
@@ -110,7 +113,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
             HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
             HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
 
-            String redirectURL = FrameworkUtils.getContextData(httpRequest).getRedirectURL();
+            AuthenticationContext authenticationContext = FrameworkUtils.getContextData(httpRequest);
+
+            if (authenticationContext != null) {
+                redirectURL = FrameworkUtils.getContextData(httpRequest).getRedirectURL();
+            } else {
+                redirectURL = null;
```

Comment:
What will be the flow if the redirectUrl is null?

Reply:
If the redirect URL is null it will again throw a different error. Since we know that there is no authentication context available in the recovery flow all recovery flows will still end in a error. We need to handle this.
https://github.com/wso2-extensions/identity-governance/blob/0fd232ceb0dabb8d4119bc4cbb92aa9dd02cb425/components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java#L165

Reply:
will be redirected to the ErrorPage [1].

[1] - https://github.com/wso2-extensions/identity-governance/blob/73fdde59c16b1cf1cd5c93d542240aa19c2338c5/components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java#L165

Reply:
I think if the redirect url is not available we will need to fallback to old implementation of calling `httpRequest.getHeader("referer")` and if that is also not available we will have to redirect to a IS error page